Hazelcast.Net/Hazelcast.Client/Statistics.cs

```diff
@@ -0,0 +1,296 @@
+﻿// Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using Hazelcast.Client.Connection;
+using Hazelcast.Client.Protocol.Codec;
+using Hazelcast.Client.Spi;
+using Hazelcast.Core;
+using Hazelcast.IO;
+using Hazelcast.Logging;
+using Hazelcast.Net.Ext;
+using Hazelcast.Security;
+using Hazelcast.Util;
+
+namespace Hazelcast.Client
+{
+    internal class Statistics
+    {
+        private static readonly ILogger Logger = Logging.Logger.GetLogger(typeof(Statistics));
+
+        private static readonly string DllVersion = VersionUtil.GetDllVersion();
+        
+        private const string NearCacheCategoryPrefix = "nc.";
+
+        private const string SinceVersionString = "3.9";
+        private static readonly int SinceVersion = VersionUtil.ParseServerVersion(SinceVersionString);
+
+        private const char StatSeparator = ',';
+        private const char KeyValueSeparator = '=';
+        private const char EscapeChar = '\\';
+        private const string EmptyStatValue = "";
+
+        private readonly HazelcastClient _client;
+        private readonly bool _enabled;
+        private readonly int _period;
+        private CancellationTokenSource _heartbeatToken;
+
+        private readonly AtomicBoolean _live = new AtomicBoolean(false);
+
+        private readonly IDictionary<string, Func<string>> _gauges = new Dictionary<string, Func<string>>();
+
+        private volatile Address _ownerAddress;
+
+        public Statistics(HazelcastClient client)
+        {
+            _client = client;
+            _enabled = EnvironmentUtil.ReadBool("hazelcast.client.statistics.enabled") ?? false;
+            _period = EnvironmentUtil.ReadInt("hazelcast.client.statistics.period.seconds") ?? 3;
```

Comment:
at Java we have this log, you may add it as well: https://github.com/hazelcast/hazelcast/blob/v3.9.2/hazelcast-client/src/main/java/com/hazelcast/client/impl/statistics/Statistics.java#L134

We also do it at start method. If not enabled, no need to print the message.

Reply:
I've added statistic enabled info log